Enalean/tuleap-oauth-plugin
# Be able to retrieve the authorization code from a Tuleap server

This change is part of [community #14019](https://tuleap.net/plugins/tracker/?aid=14019)

To minimize the size of the pull request review, only the first step of the OAuth2 Authorization Code Grant[0] request[1] flow has been implemented.
So, the access token will be retrieved in a dedicated contribution.
In this contribution the user should be able to retrieve the authoriation code from a Tuleap server by filling the form in the Global Security menu in Jenkins.
When Jenkins get the authorization code, the user will be automatically logged as "Admin" in Jenkins to avoid to be closed outside of Jenkins during tests.

How to test:
 - Create a new Oauth2 app in Tuleap.
 - Go to "Configure Global Security" menu of Jenkins
 - Choose "Tuleap Authentication", fill the form and save.
 - Log out and log in again
=> You should be redirected to the Tuleap log in page. (if you are not already logged)
 - When logged, you should be redirected to the Athorization page of Tuleap.
=> You should be redirected to Jenkins whith the authorization code (see the web console) and you should be logged as Admin.
=> If there is an error, you should be redirected to a dedicated page.

[0] https://tools.ietf.org/html/rfc6749#section-4.1
[1] https://tools.ietf.org/html/rfc6749#section-4.1.1
